From 2e2ba7c54f2ffd776dbd4724e6d9266b4fbd9c71 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Thu, 31 Oct 2019 23:10:30 +0100
Subject: [PATCH] soong: java: Specify larger heap size for metalava

filiprrs: This is needed on systems with 8GB physical ram.
Compiling using a single job is recommended.

Gegham Zakaryan <zakaryan.200@outlook.com>: adapt for R

Change-Id: I5093dcdbe384fd33f0dc0fd9b89c91f8f9fa19fd
---
 java/droiddoc.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/java/droiddoc.go b/java/droiddoc.go
index 230b1f026..7d88e2c34 100644
--- a/java/droiddoc.go
+++ b/java/droiddoc.go
@@ -1468,6 +1468,7 @@ func metalavaCmd(ctx android.ModuleContext, rule *android.RuleBuilder, javaVersi
 
 	cmd.BuiltTool(ctx, "metalava").
 		Flag(config.JavacVmFlags).
+		Flag("-J-Xmx6114m").
 		FlagWithArg("-encoding ", "UTF-8").
 		FlagWithArg("-source ", javaVersion.String()).
 		FlagWithRspFileInputList("@", srcs).
